src/mono/mono/metadata/class-init.c

```diff
@@ -436,7 +436,7 @@ mono_class_create_from_typedef (MonoImage *image, guint32 type_token, MonoError
 	MonoClass *klass, *parent = NULL;
 	guint32 cols [MONO_TYPEDEF_SIZE];
 	guint32 cols_next [MONO_TYPEDEF_SIZE];
-	guint tidx = mono_metadata_token_index (type_token);
+	gint tidx = mono_metadata_token_index (type_token);
```

Comment:
this is wrong. token indexes should be `guint32`

Reply:
Only the lower 24 bits matter for the return of `mono_metadata_token_index`. Either `guint` or `gint` should be able to preserve all possible values here.

Reply:
I don't think this is about capacity to hold onto the bits, rather conceptually about what a token type is. I tend to agree with @lambdageek that tokens should be treated as unsigned.

Reply:
I see. Will change that.

Reply:
My confusion came from passing it to other other method, where the corresponding argument type was often `int`. For this case, it was the `idx` argument of `mono_metadata_decode_row`.  I've created an issue to revisit the signedness of related arguments of various metadata methods. (https://github.com/dotnet/runtime/issues/70495)